src/AggregateRepository.EventStore/EventStoreAggregateRepository.cs

```diff
@@ -74,65 +72,70 @@ public T GetAggregateFromRepository<T>(object aggregateId, int version = int.Max
             }
 
             var streamName = StreamNameForAggregateId(aggregateId);
-            var aggregate = (T)Activator.CreateInstance(typeof(T), true);
+            var aggregate = (T)Activator.CreateInstance(typeof(T), true) !;
 
-            long sliceStart = 0;
-            var eventsCount = 0;
-            StreamEventsSlice currentSlice;
-            do
-            {
-                var sliceCount = sliceStart + ReadPageSize <= version ? ReadPageSize : version - sliceStart + 1;
-                currentSlice = _connection.ReadStreamEventsForwardAsync(streamName, sliceStart, (int)sliceCount, false).Result;
+            var readResult = _eventStoreClient.ReadStreamAsync(
+                Direction.Forwards,
+                streamName,
+                StreamPosition.Start,
+                version);
 
-                // ReSharper disable once SwitchStatementMissingSomeCases
-                switch (currentSlice.Status)
-                {
-                    case SliceReadStatus.StreamNotFound:
-                    case SliceReadStatus.StreamDeleted:
-                        throw new AggregateNotFoundException();
-                }
-
-                sliceStart = currentSlice.NextEventNumber;
+            if (readResult.ReadState.Result != ReadState.Ok)
+            {
+                throw new AggregateNotFoundException(streamName);
+            }
 
-                foreach (var evnt in currentSlice.Events)
-                {
-                    aggregate.ApplyEvent(DeserializeEvent(evnt.OriginalEvent.Metadata, evnt.OriginalEvent.Data));
-                }
+            var events = readResult.ToListAsync().Result;
```

Comment:
I've implemented a change here. But you might not like it. It introduces another use of result, which in turn required some exception handling as using results will cause all exceptions thrown to be wrapped by a System.AggregateException. When the Core library is async this can be simplified.